### DISCUSSION TITLE
How to get number of layer of an image by using imagemagick?

### DISCUSSION THREAD

**Discussion by pme9** [ImageMagick/ImageMagick#5026] [Help]
I'm reading one image using ImageMagick. And I want to get number of layer from that image by using imagemagick.
Can you please tell me how to do that?

**Comment by snibgo** [comment]
What version of IM, on what platform?

What API are you using? Command-line, or MagickWand, or what?

What do you mean by "layers"? Please link to a sample file, and say how many layers it has.

**Reply by pme9** [reply]
I'm using version 7 of IM on windows platform. 

Single file can have different layers(please find attached image).
So suppose I have an image with multiple layers then how to get it's number.

I'm not sure about which API of IM I need to use to get the count of number of layers in any image.


![Layers](https://user-images.githubusercontent.com/102242918/162747929-4273828c-4fc9-4508-a520-db3038a4eb7f.PNG)

**Comment by snibgo** [comment]
At the command line:
```
magick filename.ext -format "%N" info:
```
If the file has 3 images, the output of that command will be "3".

**Reply by pme9** [reply]
Is there any API's for this because we are using c++ in visual studio.
And we can not use command line.

**Reply by pme9** [reply]
> At the command line:
> 
> ```
> magick filename.ext -format "%N" info:
> ```
> 
> If the file has 3 images, the output of that command will be "3".

I'm reading one image using ImageMagick like below:
Magick::Image m_image;
m_image.read(filePath);

And I want to get number of layer from that image by using imagemagick.
I'm not sure about which API of IM I need to use to get the count of layers in any image.
Can you please tell me how to do that?

**Comment by snibgo** [comment]
> I'm reading one image using ImageMagick ...

You are reading only one image. So there is only one image.

This is why I asked: What do you mean by "layers"? ImageMagick sees images, not layers.
